packages/coordinator/src/services/translation/request/dispense/dispense-notification.ts

```diff
@@ -321,27 +328,75 @@ function createSuppliedLineItem(
   const fhirPrescriptionDispenseItemNumber = getPrescriptionItemNumber(fhirMedicationDispense)
   const globalIdentifier = new hl7V3.GlobalIdentifier(fhirPrescriptionDispenseItemNumber)
-  const pertientInformation2 = createSupplyPertinentInformation2(fhirMedicationDispense)
-  return new hl7V3.DispenseNotificationSuppliedLineItem(globalIdentifier, pertientInformation2)
+  const pertinentInformation2 = createSuppliedLineItemPertinentInformation2(fhirMedicationDispense)
+  return new hl7V3.DispenseNotificationSuppliedLineItem(globalIdentifier, pertinentInformation2)
 }
 
-function createSupplyPertinentInformation2(
+function createSuppliedLineItemPertinentInformation2(
   fhirMedicationDispense: fhir.MedicationDispense
 ): hl7V3.SupplyPertinentInformation2 {
-  const isNonDispensinReasonCode = getfhirStatusReasonCodeableConceptCode(fhirMedicationDispense)
-  return isNonDispensinReasonCode ?
-    createPertinentInformation2NonDispensing(isNonDispensinReasonCode) :
+  const isNonDispensingReasonCode = getFhirStatusReasonCodeableConceptCode(fhirMedicationDispense)
+  return isNonDispensingReasonCode ?
+    createPertinentInformation2NonDispensing(isNonDispensingReasonCode) :
     new hl7V3.PertinentInformation2()
 }
 
-function createPertinentInformation2NonDispensing(isNonDispensinReasonCode: fhir.Coding) {
+function createPertinentInformation2NonDispensing(isNonDispensingReasonCode: fhir.Coding) {
   const pertInformation2 = new hl7V3.NonDispensingReason(
-    isNonDispensinReasonCode.code,
-    isNonDispensinReasonCode.display
+    isNonDispensingReasonCode.code,
+    isNonDispensingReasonCode.display
   )
   return new hl7V3.PertinentInformation2NonDispensing(pertInformation2)
 }
 
+function supplyHeaderPertinentInformation2Required(fhirMedicationDispenses: Array<fhir.MedicationDispense>): boolean {
+  const nonDispensingReasonUrl = "https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionNonDispensingReason"
+
+  const nonDispensingReasonsPresent = fhirMedicationDispenses.some(
+    dispense => dispense.extension.some(
+      extension => extension.url === nonDispensingReasonUrl
+    )
+  )
+
+  const allNonDispensingReasonsPresent = fhirMedicationDispenses.every(
+    dispense => dispense.extension.some(
+      extension => extension.url === nonDispensingReasonUrl
+    )
+  )
+
+  return nonDispensingReasonsPresent && allNonDispensingReasonsPresent
+}
+
+function createSupplyHeaderPertinentInformation2(
+  fhirMedicationDispenses: Array<fhir.MedicationDispense>
+): hl7V3.PertinentInformation2NonDispensing {
+  const nonDispensingReasonUrl = "https://fhir.nhs.uk/StructureDefinition/Extension-DM-PrescriptionNonDispensingReason"
+
+  const allNonDispensingReasons = fhirMedicationDispenses.map(
+    dispense => dispense.extension.filter(
+      extension => extension.url === nonDispensingReasonUrl
+    )
+  ).flat() as Array<CodingExtension>
+
+  const nonDispensingReasonCode = allNonDispensingReasons[0].valueCoding.code
```

Comment:
is there any chance allNonDispensingReasons is an empty array and then 0 index wouldn't exist?

Reply:
There isn't, no. The preceding function (supplyHeaderPertinentInformation2Required) checks for their presence.